### ISSUE TITLE
linkExistingAddressToChain fails after createCommunity tries to assign old address to new chain

### ISSUE THREAD

**Issue by kurtisassad** [hicommonwealth/commonwealth#4666]
## Describe the bug
1. createCommunity
2. get 500 from /linkExistingAddressToChain. Fails in assertAddressOwnership with ```log.error(`Address ${address} relates to multiple profiles!`);``` conditional branch.

## Expected behavior
/linkExistingAddressToChain always returns 200 if the inputs are valid.

## "How We Fixed It"
To be determined

## Additional context
You can see examples of this in these test runs:
https://github.com/hicommonwealth/commonwealth/actions/runs/5742721590/job/15565445172

Basically, the new address that gets created somehow gets linked to an old profile, which causes this branch to trigger.
